File: fiber/internal/handlers/hello.go
package handlers

import "github.com/gofiber/fiber/v2"

type HelloRequest struct {
	Name string `json:"name"`
}

func Health(c *fiber.Ctx) error {
	return c.JSON(fiber.Map{"status": "ok"})
}

func GetHello(c *fiber.Ctx) error {
	return c.JSON(fiber.Map{"message": "Hello from Fiber!"})
}

func PostHello(c *fiber.Ctx) error {
	var req HelloRequest
	if err := c.BodyParser(&req); err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": err.Error()})
	}
	if req.Name == "" {
		req.Name = "World"
	}
	return c.JSON(fiber.Map{"message": "Hello, " + req.Name + "!"})
}

File: actix-web/src/routes/hello.rs
use actix_web::{web, HttpResponse, Responder};
use serde::{Deserialize, Serialize};

#[derive(Serialize)]
pub struct HelloResponse {
    pub message: String,
}

#[derive(Deserialize)]
pub struct HelloRequest {
    pub name: String,
}

pub async fn get_hello() -> impl Responder {
    HttpResponse::Ok().json(HelloResponse {
        message: "Hello from Actix-Web!".to_string(),
    })
}

pub async fn post_hello(body: web::Json<HelloRequest>) -> impl Responder {
    HttpResponse::Ok().json(HelloResponse {
        message: format!("Hello, {}!", body.name),
    })
}

pub async fn health() -> impl Responder {
    HttpResponse::Ok().json(serde_json::json!({ "status": "ok" }))
}
